src/Angor/Client/Pages/Invest.razor

```diff
@@ -935,4 +942,15 @@ else
     {
         showTransactionJsonModal = isVisible;
     }
+    
+    public bool CanInvest(Project project)
+    {
+        if (project == null) return false;
+
+        var now = DateTime.UtcNow;
+        if (now < project.ProjectInfo.StartDate) return false; // Investing period hasn't started
```

Comment:
no this is wrong `StartDate` is the date the project starts so it means we can only invest before the `StartDate` and we do not need to look at the `ExpiryDate`

the condition should be 

`now < project.ProjectInfo.StartDate return true`

Reply:
also add this just in case
`if (project.ProjectInfo == null) return false;`